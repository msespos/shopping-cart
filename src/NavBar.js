import { Link } from 'react-router-dom';
import "./Styles/NavBar.css"
import logo from "./imgs/logo.png"

const NavBar = () => {
  return (
    <div className='navbar'>
      <img className={logo} src={logo} width={50} />
      <Link to={'/'}>Home</Link>
      <Link to={'/shop'}>Shop</Link>
    </div>
  );
}

export default NavBar;
