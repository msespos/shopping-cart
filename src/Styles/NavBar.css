.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 30px;
  border: solid black 2px;
  padding: 20px;
}

a {
  text-decoration: none;
  color: black;
}

em {
  font-family: 'Courier New', Courier, monospace;
  font-size: medium;
}

.note {
  margin: 30px;
}
