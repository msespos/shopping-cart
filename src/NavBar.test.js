import { render, screen } from "@testing-library/react";
import { HashRouter } from "react-router-dom";
import NavBar from "./NavBar";
import '@testing-library/jest-dom';

describe("App component", () => {
  it("renders Home and Shop in the navbar", () => {
    const { getByText } = render(<NavBar />, {wrapper: HashRouter});
    expect(getByText("Home")).toBeInTheDocument();
    expect(getByText("Shop")).toBeInTheDocument();
  });

  it("renders the component", () => {
    const component = render(<NavBar />, {wrapper: HashRouter});
    expect(component).toMatchSnapshot()
  });

  it("has the correct source for the logo image in the navbar", () => {
    render(<NavBar />, {wrapper: HashRouter});
    const logoImage = screen.getByAltText(/VAA Logo/i);
    const imageSource = "logo.png";
    expect(logoImage).toHaveAttribute("src", imageSource);
  });
});
