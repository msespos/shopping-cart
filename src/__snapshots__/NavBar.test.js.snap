// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component includes some of the blurb in the navbar note: 
      this is not an actual store.
     1`] = `
<DocumentFragment>
  <div>
    <div
      class="note"
    >
      <em>
        Note: this is not an actual store. It's simply a site created for learning purposes. I'm not actually selling anything, least of all Autechre's music!
        <br />
        But you should definitely go listen to Autechre right now!
      </em>
    </div>
    <div
      class="navbar"
    >
      <a
        href="#/"
      >
        <img
          alt="VAA logo"
          src="logo.png"
          width="50"
        />
      </a>
      <a
        href="#/"
      >
        Home
      </a>
      <a
        href="#/shop"
      >
        Shop
      </a>
    </div>
  </div>
</DocumentFragment>
`;
